runtime/lua/man.lua

```diff
@@ -305,7 +305,7 @@ local function matchstr(text, pat_or_re)
     return
   end
 
-  return text:sub(vim.str_utfindex(text, s) + 1, vim.str_utfindex(text, e))
+  return text:sub(vim.str_utfindex(text, 'utf-32', s) + 1, vim.str_utfindex(text, 'utf-32', e))
```

Comment:
starting to think the encoding param should be the first param. WDYT @lewis6991 ?
```suggestion
  return text:sub(vim.str_utfindex('utf-32', text, s) + 1, vim.str_utfindex('utf-32', text, e))
```

Reply:
My intuition prefers that the main subject is the first argument, and subsequent arguments are inputs regarding the subject. e.g. `string.gsub(text, pattern, repl, n)`.

Or another way to think of this. If `str_utf_index` was a string method it would look like `text:utf_index('utf-32', index)`.

So if anything I'd prefer the encoding to be last, however legacy prevents us doing that.